docs/tutorials/generate-custom-test-data.rst

```diff
@@ -0,0 +1,125 @@
+.. _ksql_generate-custom-test-data:
+
+Generate Custom Test Data by Using the ksql-datagen tool
+========================================================
+
+You can use the ``ksql-datagen`` command-line tool to generate test data that
+complies with a custom schema that you define.
+
+To generate test data, create an Apache Avro schema and pass it to 
+``ksql-datagen``. This generates random data according to the schema you
+provide.
+
+**Prerequisites:** 
+
+- :ref:`Confluent Platform <installation>` is installed and running.
+  This installation includes a Kafka broker, KSQL, |c3-short|, |zk|,
+  Schema Registry, REST Proxy, and Kafka Connect.
+- If you installed |cp| via TAR or ZIP, navigate to the installation
+  directory. The paths and commands used throughout this tutorial assume
+  that you're in this installation directory.
+- Java: Minimum version 1.8. Install Oracle Java JRE or JDK >= 1.8 on your
+  local machine.
+
+Start a Kafka cluster, including |zk|, a Kafka broker, and |sr|. KSQL queries
+messages from this Kafka cluster. KSQL is installed in the |cp| by default.
+
+Define a custom schema
+----------------------
+
+In this example, you register a custom Avro schema to the |sr|. The schema is
+named `impressions.avro 
+<https://github.com/apurvam/streams-prototyping/blob/master/src/main/resources/impressions.avro>`_, 
+and it represents ads delivered to users.
+
+#. Download ``impressions.avro`` and copy it to a convenient directory. It's used
+   by ``ksql-datagen`` when you start generating test data.
+
+   .. code:: bash
+
+      curl https://raw.githubusercontent.com/apurvam/streams-prototyping/master/src/main/resources/impressions.avro > impressions.avro
+
+#. Use the ``curl`` command to post the ``impressions-key`` schema to the |sr|.
+
+   .. code:: bash
+
+      curl -X POST -H "Content-Type: application/vnd.schemaregistry.v1+json" \
+      --data '{"schema": "{\"namespace\":\"streams\",\"name\":\"impressions\",\"type\":\"string\"}"}' \

```

Comment:
You don't need to register the schema in SchemaRegistry. The data-gen will automatically register the schema.